gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaPartition.java

```diff
@@ -31,6 +31,7 @@
   private final int id;
   private final String topicName;
   private KafkaLeader leader;
+  private int hashCode;
```

Comment:
Hmm. I can, but is there a significant benefit to using auto-generated hashCode? The only change I am making here is to cache the hashCode to avoid re-computing it every time.